ui/apps/platform/src/Containers/Collections/CollectionsFormPage.tsx

```diff
@@ -108,46 +112,83 @@ function CollectionsFormPage({
     }
 
     function onSubmit(collection: Collection): Promise<void> {
-        if (pageAction.type === 'view') {
-            // Logically should not happen, but just in case
-            return Promise.reject();
-        }
+        setSaveError(undefined);
+
+        return new Promise((resolve, reject) => {
+            if (pageAction.type === 'view') {
+                // Logically should not happen, but just in case
+                return reject(new Error('A Collection form has been submitted in read-only view'));
+            }
+            const isEmptyCollection =
+                Object.values(collection.resourceSelector).every(({ type }) => type === 'All') &&
+                collection.embeddedCollectionIds.length === 0;
+
+            if (isEmptyCollection) {
+                return reject(new Error('Cannot save an empty collection'));
```

Comment:
A question for understanding, not a change request: unlike some other validation errors, yup validation might in theory be able to detect empty collection?

Reply:
I agree that yup could do this in theory, since we can determine an empty collection 100% on the client. In practice it didn't seem readily apparent on how to accomplish this by reading through the `yup` docs and I went with this solution since the empty checks were fairly straightforward. I think part of the complication is that the validation would apply both to `values.embeddedCollectionIds` (array) and `values.resourceSelectors` (Record), the latter of which seemed to need to have errors specified for each of the `Deployment`, `Namespace`, and `Cluster` property keys.

I'll make a note to look into this again though and see if there is a clean idiomatic way to get the same behavior via `yup`.